src/test/kotlin/me/serce/solidity/lang/core/resolve/SolEventResolveTest.kt

```diff
@@ -61,22 +67,121 @@ class SolEventResolveTest : SolResolveTestBase() {
     RecursionManager.disableMissedCacheAssertions(testRootDisposable)
     checkByCode(
       """
+         pragma solidity ^0.8.26;
+         
         contract B {
             event B();
+                //x  
                    
             function close() public {
                 emit B();
                    //^
             }
             
-            function B() {
-                   //x
+            function B() public {
+            }
+        }
+    """
+    )
+  }
+
+  fun testContractTheSameNameAsEvent2() {
+    // A recursion guard check is disabled to prevent c.i.o.u.RecursionManager.CachingPreventedException
+    // from being thrown. The case highlights another issue that's worth fixing.
```

Comment:
I haven't tried to dive deeper yet, but is this issue only occurring on newly added targets? I'm a bit concerned about the performance implications of any cases where caching doesn't work as expected.

Reply:
I was unable to reproduce any problem using this test even with `RecursionManager.disableMissedCacheAssertions` commented out

Reply:
should I remove the TODO ?